src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_k8s_linux.go

```diff
@@ -52,16 +60,26 @@ func executeK8SBundle(ctx context.Context, bp bundleParams) error {
 		saveInterrupts(ps),
 		saveResourceUsageData(ps, bp.cfg),
 		saveNTPDrift(ps),
-		savePrometheusMetrics(ctx, ps, bp.admin),
 		saveDiskUsage(ctx, ps, bp.cfg),
 		saveControllerLogDir(ps, bp.cfg, bp.controllerLogLimitBytes),
 	}
+
+	adminAddresses, err := adminAddressesFromK8S(ctx)
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("skipping admin API calls, unable to get admin API addresses: %v", err))
+	} else {
+		steps = append(steps, []step{
+			saveClusterAdminAPICalls(ctx, ps, bp.fs, bp.cfg, adminAddresses),
+			saveSingleAdminAPICalls(ctx, ps, bp.fs, bp.cfg, adminAddresses),
+		}...)
+	}
 	for _, s := range steps {
 		grp.Go(s)
 	}
 
-	errs := grp.Wait()
-	if errs != nil {
+	stepErrs := grp.Wait()
+	if stepErrs != nil {
+		errs = multierror.Append(errs, stepErrs.ErrorOrNil())
```

Comment:
Why the logic change here? ErrorOrNil -- we already know this is an error -- I think this is the same as before?

Reply:
Because now we have `errs` that is the final list of errors, and stepErrs that is the errors from steps. We can have now errors outside of the step execution, such as failures to retrieve the adminAPI addresses from the k8s API.

The logic change is now:
- We create a slice of `steps`
- We try to get the adminAddresses from K8S
    - If we have an error, we add that to `errs` and skip the steps
    - If not, add the adminAPI steps to the list.
- Run the steps concurrently, get the errors and append to the final list (`errs`)

Reply:
I see now, thanks!